app/javascript/controllers/organization_select_controller.js

```diff
@@ -0,0 +1,184 @@
+import { Controller } from '@hotwired/stimulus'
+import fuzzysort from 'fuzzysort'
+
+export default class extends Controller {
+  static targets = [
+    'dropdown',
+    'menu',
+    'search',
+    'organization',
+    'wrapper',
+    'field',
+  ]
+  static values = {
+    state: Boolean,
+  }
+
+  connect() {
+    const organizations = {}
+
+    const open = () => {
+      // eslint-disable-next-line no-undef
+      $(this.menuTarget).slideDown()
+      this.searchTarget.style.display = 'block'
+      this.dropdownTarget.style.display = 'none'
+      this.searchTarget.select()
+    }
+
+    const close = () => {
+      // eslint-disable-next-line no-undef
+      $(this.menuTarget).slideUp()
+      this.searchTarget.style.display = 'none'
+      this.dropdownTarget.style.display = 'flex'
+    }
+
+    const filter = async () => {
+      const orgValues = Object.values(organizations)
+
+      const start = performance.now()
+      const result = fuzzysort.go(this.searchTarget.value, orgValues, {
+        keys: ['name', 'id'],
+        all: false,
+        threshold: -500000,
+        limit: 50,
+      })
+      const end = performance.now()
+
+      firstOrganization = result[0]?.obj
+
+      const shown =
+        result.length > 0
+          ? result.map(r => r.obj.organization)
+          : orgValues
+              .sort((a, b) => a.index - b.index)
+              .map(o => o.organization)
+              .slice(0, 50)
+      const hidden = this.organizationTargets.filter(el => !shown.includes(el))
+      console.log('Search took', end - start, 'ms')
+      const renderStart = performance.now()
+
+      for (const element of shown) {
+        ;(async () => {
+          element.parentElement.appendChild(element)
+          element.style.display = 'block'
+        })()
+      }
+
+      for (const element of hidden) {
+        ;(async () => {
+          element.style.display = 'none'
+        })()
+      }
+
+      const renderEnd = performance.now()
+
+      console.log('Render took', renderEnd - renderStart, 'ms')
+    }
+
+    for (const organization of this.organizationTargets) {
+      const { name, id, fee, index } = organization.dataset
+      const button = organization.children[0]
+      const select = () => {
+        const oldFieldValue =
+          organizations[this.dropdownTarget.children[1].value]
+        if (oldFieldValue) {
+          Object.assign(oldFieldValue.button.style, {
+            backgroundColor: 'unset',
+            color: 'unset',
+          })
+          oldFieldValue.button.children[1].style.color = ''
+        }
+
+        Object.assign(button.style, {
+          backgroundColor: 'var(--info)',
+          color: 'white',
+        })
+        button.children[1].style.color = 'white'
```

Comment:
Is there a reason why we aren't defining these styles in a class and then toggling the class?

Reply:
It's 2 styles, and I think we're trying to shift towards tailwind instead of .scss files.

Reply:
Imo they should be in classes - I feel like that's cleaner and will make it easier to change down the line. Plus then you can just add and remove the class, not use unset or empty strings